### PR TITLE
Various bugfixes for dev-util/bcc and dev-util/bpftrace

### PR DESCRIPTION
**Pull request by jsmolic**
I've added some fixes for these package, more details are included in the commit descriptions.
@chutz if don't mind, I'd add myself as co-maintainer for these packages, I'm using some bpf and generally have interest in this so I'd like to help out with maintenance, version bumps, bugfixes, etc.

### CODE REVIEW

File: dev-util/bcc/bcc-0.22.0-r1.ebuild [hunk 1]
```diff
@@ -19,15 +18,17 @@ KEYWORDS="~amd64 ~arm64 ~x86"
...
 RDEPEND="
-	~dev-libs/libbpf-0.3:=[static-libs(-)]
-	>=sys-kernel/linux-headers-4.14
 	>=dev-libs/elfutils-0.166:=
-	<=sys-devel/clang-13:=
-	<=sys-devel/llvm-13:=[llvm_targets_BPF(+)]
-	lua? ( ${LUA_DEPS} )
+	>=dev-libs/libbpf-0.5.0:=[static-libs(-)]
+	>=sys-kernel/linux-headers-5.13
+	<=sys-devel/clang-14:=
```
Comment (thesamesam) [inline dev-util/bcc/bcc-0.22.0-r1.ebuild:28]: I prefer:
```suggestion
	<sys-devel/clang-$((${LLVM_MAX_SLOT} + 1)):=
```
.. and so on. Then just define LLVM_MAX_SLOT at the very top of the ebuild above inherit for cleanliness.

File: dev-util/bcc/bcc-0.22.0-r1.ebuild [hunk 2]
```diff
@@ -61,7 +58,7 @@ pkg_pretend() {
 }
 
 pkg_setup() {
-	LLVM_MAX_SLOT=11 llvm_pkg_setup
+	LLVM_MAX_SLOT=13 llvm_pkg_setup
```
Comment (thesamesam) [inline dev-util/bcc/bcc-0.22.0-r1.ebuild:61]: I think we should just do this above inherit (see other cmoment).

File: dev-util/bcc/bcc-0.22.0-r1.ebuild [hunk 3]
```diff
@@ -72,6 +69,13 @@ src_prepare() {
 	bpf_link_path="$(realpath --relative-to="${S}/src/cc/libbpf" /usr/include/bpf)" || die
 	ln -sfn "${bpf_link_path}" src/cc/libbpf/include || die
 
+	# bug 811288
+	local script scriptname
+	for script in $(find tools/old -type f -name "*.py"); do
```
Comment (thesamesam) [inline dev-util/bcc/bcc-0.22.0-r1.ebuild:74]: Nits would be that we should have || die on the find (inside the subshell) and on basename too.

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild [hunk 4]
```diff
@@ -70,19 +62,23 @@ pkg_pretend() {
...
 		-DBUILD_FUZZ:BOOL=$(usex fuzzing)
+		-DENABLE_MAN:BOOL=OFF
 	)
 
 	cmake_src_configure
 }
+
+src_install() {
+	cmake_src_install
+	dostrip -x /usr/bin/bpftrace
```
Comment (thesamesam) [inline dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:82]: whack in a bug ref

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild [hunk 5]
```diff
@@ -70,19 +62,23 @@ pkg_pretend() {
...
+	LLVM_MAX_SLOT=13 llvm_pkg_setup
 }
 
 src_configure() {
 	local -a mycmakeargs=(
 		-DSTATIC_LINKING:BOOL=OFF
+		-DBUILD_SHARED_LIBS:=OFF
 		-DBUILD_TESTING:BOOL=OFF
 		-DBUILD_FUZZ:BOOL=$(usex fuzzing)
+		-DENABLE_MAN:BOOL=OFF
```
Comment (thesamesam) [inline dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:74]: any reason?

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild [hunk 6]
```diff
@@ -70,19 +62,23 @@ pkg_pretend() {
...
-
-src_prepare() {
-	cmake_src_prepare
+	LLVM_MAX_SLOT=13 llvm_pkg_setup
 }
 
 src_configure() {
 	local -a mycmakeargs=(
 		-DSTATIC_LINKING:BOOL=OFF
+		-DBUILD_SHARED_LIBS:=OFF
```
Comment (thesamesam) [inline dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:71]: worth explaining why if we have a bug

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild [hunk 7]
```diff
@@ -3,59 +3,51 @@
...
 
-COMMON_DEPEND="
+RDEPEND="
 	dev-libs/libbpf:=
 	>=dev-util/bcc-0.13.0:=
 	dev-util/systemtap
-	>=sys-devel/llvm-6:=[llvm_targets_BPF(+)]
-	<=sys-devel/llvm-13:=[llvm_targets_BPF(+)]
-	<=sys-devel/clang-13:=
+	<=sys-devel/clang-14:=
```
Comment (thesamesam) [inline dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:25]: same dep changes as mentioned on other

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild [hunk 8]
```diff
@@ -3,59 +3,51 @@
...
+	<=sys-devel/llvm-14:=[llvm_targets_BPF(+)]
 	sys-libs/binutils-libs:=
 	virtual/libelf
```
Comment (thesamesam) [inline dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:28]: :=

**Review by thesamesam — CHANGES_REQUESTED** [review]

**Comment by thesamesam** [comment]
Thanks by the way! Great work and nice bug fixes.

File: dev-util/bcc/bcc-0.22.0-r1.ebuild (hunk 1 above)
Comment (jsmolic) [reply dev-util/bcc/bcc-0.22.0-r1.ebuild:28]: Oh yes, this looks cleaner, thanks

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild (hunk 5 above)
Comment (jsmolic) [reply dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:74]: Hmm I'll check if we can switch it to on (this was same as in previous versions, but GH interface showed it as being added in this revision)

Comment (thesamesam) [reply dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:74]: Ah, ok, don't treat it as a blocker for merging then given some of these fixes are quite important

File: dev-util/bcc/bcc-0.22.0-r1.ebuild (hunk 3 above)
Comment (jsmolic) [reply dev-util/bcc/bcc-0.22.0-r1.ebuild]: Done

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild (hunk 6 above)
Comment (jsmolic) [reply dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:74]: Added bug ref

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild (hunk 8 above)
Comment (jsmolic) [reply dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild]: :+1:

File: dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild (hunk 5 above)
Comment (jsmolic) [reply dev-util/bpftrace/bpftrace-0.14.0-r3.ebuild:77]: okay

File: dev-util/bcc/bcc-0.22.0-r1.ebuild [hunk 9]
```diff
@@ -25,8 +26,8 @@ RDEPEND="
 	>=dev-libs/elfutils-0.166:=
 	>=dev-libs/libbpf-0.5.0:=[static-libs(-)]
 	>=sys-kernel/linux-headers-5.13
-	<=sys-devel/clang-14:=
-	<=sys-devel/llvm-14:=[llvm_targets_BPF(+)]
+	<=sys-devel/clang-$((${LLVM_MAX_SLOT} + 1)):=
```
Comment (thesamesam) [inline dev-util/bcc/bcc-0.22.0-r1.ebuild:29]: Note that we need <, not <=, if we're doing ```... + 1```.

The reason being that for say, LLVM_MAX_SLOT=13, we want to allow 13.0.1, and so on, but not 14.

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: dev-util/bcc/bcc-0.22.0-r1.ebuild (hunk 9 above)
Comment (jsmolic) [reply dev-util/bcc/bcc-0.22.0-r1.ebuild]: yes, overlooked that `<=` was still present

**Review by thesamesam — APPROVED** [review]

**Pull request closed without merging**